src/stdatm/speed_parameters.py

```diff
@@ -10,3 +11,264 @@
+    """
+
+    :param true_airspeed: in m/s
+    :param kinematic_viscosity: in m**2/s
+    :return: unitary_reynolds in 1/m
+    """
+    unitary_reynolds = true_airspeed / kinematic_viscosity
+    return unitary_reynolds
+
+
+# DYNAMIC PRESSURE =================================================================
+def compute_dynamic_pressure(mach, pressure):
+    """
+
+    :param mach: no unit
+    :param pressure: in Pa
+    :return: dynamic pressure in Pa
+    """
+    true_airspeed = 0.7 * mach**2 * pressure
+    return true_airspeed
+
+
+# IMPACT PRESSURE =================================================================
+def _compute_subsonic_impact_pressure(mach, pressure):
+    return pressure * ((1.0 + 0.2 * mach**2) ** 3.5 - 1.0)
+
+
+def _compute_supersonic_impact_pressure(mach, pressure):
+    # Rayleigh law
+    # https://en.wikipedia.org/wiki/Rayleigh_flow#Additional_Rayleigh_Flow_Relations
+    return pressure * (166.92158 * mach**7 / (7 * mach**2 - 1.0) ** 2.5 - 1.0)
+
+
+@singledispatch
+def compute_impact_pressure(mach, pressure):
+    """
+
+    :param mach: no unit
+    :param pressure: in Pa
+    :return: impact pressure in Pa
+    """
+    # Implementation for numpy arrays
+    mach = np.asarray(mach)
+    idx_subsonic = mach <= 1.0
+    idx_supersonic = np.logical_not(idx_subsonic)
+
+    if np.shape(pressure) != np.shape(mach):
+        pressure = np.broadcast_to(pressure, np.shape(mach))
+    else:
+        pressure = np.asarray(pressure)
+
+    impact_pressure = np.empty_like(mach)
+    impact_pressure[idx_subsonic] = _compute_subsonic_impact_pressure(
+        mach[idx_subsonic], pressure[idx_subsonic]
+    )
+    impact_pressure[idx_supersonic] = _compute_supersonic_impact_pressure(
+        mach[idx_supersonic], pressure[idx_supersonic]
+    )
+    return impact_pressure
+
+
+@compute_impact_pressure.register
+def _(mach: Number, pressure: Number):
+    # Implementation for floats
+    if mach <= 1.0:
+        return _compute_subsonic_impact_pressure(mach, pressure)
+    else:
+        return _compute_supersonic_impact_pressure(mach, pressure)
+
+
+# CALIBRATED AIRSPEED =================================================================
+#         Computation is done using Eq. 3.16 and 3.17 from:
+#         Gracey, William (1980), "Measurement of Aircraft Speed and Altitude",
+#         NASA Reference Publication 1046.
+#         https://apps.dtic.mil/sti/pdfs/ADA280006.pdf
+
+# Pre-calculation of some equation constants for sake of speed.
+GAMMA_MINUS_ONE_OVER_GAMMA = (1.4 - 1.0) / 1.4
+GAMMA_MINUS_ONE_OVER_TWO_GAMMA = (1.4 - 1.0) / 2.8
+COEFF_1 = 6.0**2.5 * 1.2**3.5
+
+
+def _compute_cas_low_speed(impact_pressure, sea_level_pressure, sea_level_speed_of_sound):
+    return sea_level_speed_of_sound * np.sqrt(
+        5.0 * ((impact_pressure / sea_level_pressure + 1.0) ** GAMMA_MINUS_ONE_OVER_GAMMA - 1.0)
+    )
+
+
+def _equation_cas_high_speed(cas, impact_pressure, sea_level_pressure, sea_level_speed_of_sound):
```

Comment:
Why not `_equation_cas_supersonic` as is done for the impact pressure ? Same for low_speed and subsonic

Reply:
Because what drives the choice of the equation is the comparison between the resulting CAS and the speed of sound at sea level, which does not match the subsonic/supersonic limit.
I'll add comments in these 2 functions to make that clearer.

Reply:
Sorry, I didn't check where it was used.